pkg/gpu/nvidia/manager.go

```diff
@@ -109,6 +110,24 @@ func (config *GPUConfig) AddDefaultsAndValidate() error {
 	return nil
 }
 
+
+func (config *GPUConfig) AddHealthCriticalXid() error {
+	xidConfig := os.Getenv("XID_CONFIG")
+	if len(xidConfig) == 0 {
+		glog.Infof("There is no Xid config specified ")
+		return nil
+	}
+
+	glog.Infof("Detected HealthCriticalXid")
+	xidStrs := strings.Split(xidConfig, ",")
+        xidArry := make([]int, len(xidStrs))
+        for i := range xidArry {
+	    xidArry[i], _ = strconv.Atoi(xidStrs[i])
```

Comment:
Otherwise, consider the case if someone mistakenly enters something like "1, 2, x". The string "x" would fail to convert. Even worse, the array is initialized as an array of 0's, so now 0 is treated as a health critical xid - which could cause some very bad consequences.

Reply:
+1